### DISCUSSION TITLE
Automatically add `getId` prop to dynamic routes

### DISCUSSION THREAD

**Discussion by nandorojo** [expo/router#97] [Ideas]
For dynamic paths like `/artists/[id]`, we should add `getId` to the `Screen`:

```tsx
const dynamicPath = 'id'

return <Screen getId={({ params }) => params?.[dynamicPath]} />
```

The [`getId`](https://reactnavigation.org/docs/screen/#getid) function lets you push the same screen multiple times, if the ID differs (for example, pushing multiple artists on top of each other).

**Comment by EvanBacon** [comment]
Good idea, this has been added in `expo-router@0.0.33`

**Reply by nandorojo** [reply]
Related: https://twitter.com/satya164/status/1567162295767207938?s=20&t=wzLSKI6opRrOjKGahJJw_w

`navigate` no longer goes back to a specific screen. Rather, you need to use `popTo` for that, which is better IMO. https://github.com/react-navigation/react-navigation/pull/10818

**Reply by nandorojo** [reply]
It appears that `getId` is actually preserving the previous behavior, which is not exactly desired. Previously I used `getId` so that we could push multiple `Artist` screens on top of each other. However, given that it now is no longer needed to keep pushing the same screen over and over, we may no longer need `getId` here. 

So once expo-router upgrades to React Navigation v7, I think we can remove this option.
